detector.go

```diff
@@ -27,6 +28,8 @@ var errBuildpack = errors.New("buildpack(s) failed with err")
 var errInconsistentVersion = `buildpack %s has a "version" key that does not match "metadata.version"`
 var errDoublySpecifiedVersions = `buildpack %s has a "version" key and a "metadata.version" which cannot be specified together. "metadata.version" should be used instead`
 var warnTopLevelVersion = `Warning: buildpack %s has a "version" key. This key is deprecated in build plan requirements in buildpack API 0.3. "metadata.version" should be used instead`
+var errInvalidRequirementsBuildpack = `priviledged buildpack %s has defined "requires", which is not allowed.`
+var errInvalidProvidesBuildpack = `buildpack %s has defined "provide" with "mixin = true", which is not allowed.`
```

Comment:
Check out https://github.com/buildpacks/lifecycle/commit/2f74559b2af29ad09e24cbae5a812c0fe78dcb51 which introduces a different pattern for the custom errors.